packages/ckeditor5-widget/tests/verticalnavigation.js

```diff
@@ -0,0 +1,1379 @@
+					editor.editing.view.document.fire( 'keydown', upArrowDomEvtDataStub );
+
+					sinon.assert.notCalled( upArrowDomEvtDataStub.preventDefault );
+					sinon.assert.notCalled( upArrowDomEvtDataStub.stopPropagation );
+				} );
+
+				it( 'should not prevent default on forward navigation', () => {
+					editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+					sinon.assert.notCalled( downArrowDomEvtDataStub.preventDefault );
+					sinon.assert.notCalled( downArrowDomEvtDataStub.stopPropagation );
+				} );
+			} );
+		} );
+
+		it( 'should integrate with the blockquote (forward navigation)', () => {
+			setModelData( model,
+				'<blockQuote>' +
+					'<paragraph>f[]oo</paragraph>' +
+				'</blockQuote>' +
+				'<horizontalLine></horizontalLine>' +
+				'<paragraph>bar</paragraph>'
+			);
+
+			editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+			sinon.assert.calledOnce( downArrowDomEvtDataStub.preventDefault );
+			sinon.assert.calledOnce( downArrowDomEvtDataStub.stopPropagation );
+
+			assertEqualMarkup( getModelData( model ),
+				'<blockQuote>' +
+					'<paragraph>foo[]</paragraph>' +
+				'</blockQuote>' +
+				'<horizontalLine></horizontalLine>' +
+				'<paragraph>bar</paragraph>'
+			);
+		} );
+
+		it( 'should integrate with the blockquote (backward navigation)', () => {
+			setModelData( model,
+				'<paragraph>foo</paragraph>' +
+				'<horizontalLine></horizontalLine>' +
+				'<blockQuote>' +
+					'<paragraph>ba[]r</paragraph>' +
+				'</blockQuote>'
+			);
+
+			editor.editing.view.document.fire( 'keydown', upArrowDomEvtDataStub );
+
+			sinon.assert.calledOnce( upArrowDomEvtDataStub.preventDefault );
+			sinon.assert.calledOnce( upArrowDomEvtDataStub.stopPropagation );
+
+			assertEqualMarkup( getModelData( model ),
+				'<paragraph>foo</paragraph>' +
+				'<horizontalLine></horizontalLine>' +
+				'<blockQuote>' +
+					'<paragraph>[]bar</paragraph>' +
+				'</blockQuote>'
+			);
+		} );
+	} );
+
+	describe( 'with selection inside nested content editable', () => {
+		let styleElement;
+
+		beforeEach( () => {
+			styleElement = global.document.createElement( 'style' );
+			styleElement.type = 'text/css';
+			styleElement.appendChild( global.document.createTextNode(
+				`
+					* {
+						font-size: 12px !important;
+						font-family: serif !important;
+						margin: 0 !important;
+						padding: 0 !important;
+						border: 0 !important
+					}
+					td { width: 30px !important; }
+					tr:nth-child(2) td:nth-child(2) { width: 300px !important; }
+					`
+			) );
+			global.document.querySelector( 'head' ).appendChild( styleElement );
+		} );
+
+		afterEach( () => {
+			styleElement.remove();
+		} );
+
+		describe( 'simple cell text content', () => {
+			describe( 'with collapsed selection', () => {
+				beforeEach( () => {
+					setModelData( model, modelTable( [
+						[ '00', '01', '02' ],
+						[ '10', 'foo[]bar', '12' ],
+						[ '20', '21', '22' ]
+					] ) );
```

Comment:
I think that a general (defined ad-hoc) nested editable would be better than a specialised table construct.

something like:

```html
<paragraph>Foo bar</paragraph>
<editable>before</editable>
<editable>foo[]bar</editable>
<editable>after</editable>
```

or, if needed the `<editable>` might be two-level deep (like image): `<block><editable></editable></block>`.

The reason for this is that not everybody must know tables internals and it's nuisances to work with a general widget feature.

Especially because the test suite is for "with selection inside nested content editable" but sub-suite is called "simple cell text content".

Reply:
I was to lazy. I'll change this :)